virttest/qemu_devices/qcontainer.py

```diff
@@ -1061,16 +1061,26 @@ def pflash_handler(firmware_name, cmd):
                 first_image = images[0]
                 img_params = params.object_params(first_image)
                 img_params["backing_chain"] = "no"
-                img_info = json.loads(qemu_storage.QemuImg(img_params,
-                                                           current_data_dir,
-                                                           first_image).info(True, "json"))
-                img_path, img_name = os.path.split(img_info.get("filename"))
-
-                pflash_vars_name = "%s_%s_VARS.fd" % (self.vmname, img_name)
-                pflash_vars_path = os.path.join(img_path, pflash_vars_name)
-                # When image has backing files, treat it as a temporary image
-                if "backing-filename" in img_info:
-                    self.temporary_image_snapshots.add(pflash_vars_path)
+                img_obj = storage.QemuImg(img_params, current_data_dir, first_image)
```

Comment:
If i understand correctly,  `qemu_storage.QemuImg` support the method `is_remote_image` too, as
https://github.com/avocado-framework/avocado-vt/blob/abf6e30858bc26f73570ed2ffb86672a81cdb170/virttest/qemu_storage.py#L396

Can combine with L1074 i think.